### PR TITLE
Avoid setting zero and commit if split fails in extent_recycle.

### PR DESCRIPTION
**Pull request by interwq**
Avoid setting zero and commit in extent_recycle_extract.

### CODE REVIEW

**Comment by interwq** [comment]
@davidtgoldblatt: updated the diff after the offline discussion.

**Comment by davidtgoldblatt** [comment]
Why should `extent_recycle_extract` be setting zero at all? It seems to me that we should assign to *zero at most once, once we're assured of success in the main `extent_recycle`.

**Comment by interwq** [comment]
That's valid. Didn't realized extent_recycle_extract was only used in extent_recycle. I'll refactor that part then.

**Comment by interwq** [comment]
@davidtgoldblatt: refactored both zero and commit to be not included in extent_recycle_extract.

File: src/extent.c [hunk 1]
```diff
@@ -1021,23 +1014,25 @@ extent_recycle(tsdn_t *tsdn, arena_t *arena, extent_hooks_t **r_extent_hooks,
...
-		*commit = true;
-	}
 
 	extent = extent_recycle_split(tsdn, arena, r_extent_hooks, rtree_ctx,
 	    extents, new_addr, size, pad, alignment, slab, szind, extent,
 	    growing_retained);
 	if (extent == NULL) {
 		return NULL;
 	}
+	if (extent_committed_get(extent)) {
```
Comment (davidtgoldblatt) [inline src/extent.c]: We can still fail here, no?

Comment (davidtgoldblatt) [reply src/extent.c]: It seems line 1046 (which, weirdly, I don't seem able to comment on) is the earliest place we should set *commit and *zero

Comment (interwq) [reply src/extent.c]: Ah good catch. Just fixed.

**Pull request merged**